2017/1/day1.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+func intFromRune(r rune) (int64, error) {
+	s_r := string(r)
+	v, err := strconv.ParseInt(s_r, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return v, nil
+}
+
+func nextPart2(i int, input string) int {
+	// Last char is a line feed
+	real_len := len(input) - 1
+	// The length of the input will always be divisible by 2
+	half := real_len / 2
+	return (i + half) % real_len
+}
+
+func nextPart1(i int, input string) int {
+	// Last char is a line feed
+	if i == len(input)-2 {
+		return 0
+	} else {
+		return i + 1
+	}
+}
+
+func main() {
+	dat, err := ioutil.ReadFile("input.txt")
+	if err != nil {
+		fmt.Printf("Couldn't read file: %v\n", err)
+		os.Exit(1)
```

Comment:
Since you might have to do this a lot it might save you some typing to use `log.Fatalf`. 

The log library also automatically adds newlines so you won't have to type those either.

Reply:
question re the Go logging lib: is it like python, where if you don't explicitly set logs to a certain level, they don't appear in the terminal?

do i have to initialize logging in order to use Fatalf?

Reply:
The logging library in the standard library doesn't have log levels. It should should work out of the box though!

Reply:
ah that's right! i thought i had the experience recently of using the logging lib and nothing was logged, so I assumed it was logging at the wrong level or something but obviously that wasn't it - and then I totally failed to look into it